### PR TITLE
Correct plugin logging

### LINKED ISSUE #3188: Logs from plugins are always shown as debug logs
## Describe the bug
Logs from plugins are always shown as debug logs, even if it is info, warning or error logs.

## To Reproduce

1. Ensure that Docker is not running
1. Run `porter list`
1. See that the logs output doesn't contain the error log message, "Docker is not available", written by the by the plugin. Only the RPC error is returned.
1. Run `PORTER_LOGS_STRUCTURED=true porter list --verbosity debug`
1. See that the logs output does contain the error log message, "Docker is not available", written by the by the plugin. But it is written with the debug log level.

## Expected behavior
Information, warning and error logs from plugins should be written with the correct log level.

## Porter Command and Output
```
PORTER_LOGS_STRUCTURED=true porter list
2024-07-13T23:00:18.371+0200    error   rpc error: code = Unknown desc = Docker is not available
2024-07-13T23:00:18.371+0200    error   could not list installations: could not read storage schema document: rpc error: code = Unknown desc = Docker is not available
2024-07-13T23:00:18.371+0200    error   could not list installations: could not read storage schema document: rpc error: code = Unknown desc = Docker is not available
```

```
PORTER_LOGS_STRUCTURED=true porter list --verbosity debug
2024-07-13T22:57:36.433+0200    debug   Checking database schema
2024-07-13T22:57:36.433+0200    debug   Selected configured plugin      {"plugin-key": "mongodb-docker"}
2024-07-13T22:57:36.434+0200    debug   Connecting to plugin    {"plugin-command": "/home/kim/.porter/porter plugin run storage.porter.mongodb-docker"}
2024-07-13T22:57:36.434+0200    debug   starting plugin
2024-07-13T22:57:36.434+0200    debug   plugin started
2024-07-13T22:57:36.434+0200    debug   waiting for RPC address
2024-07-13T22:57:36.451+0200    debug   using plugin
2024-07-13T22:57:36.452+0200    debug   plugin address
2024-07-13T22:57:36.485+0200    debug   Docker is not available
2024-07-13T22:57:36.485+0200    debug   Docker is not available
2024-07-13T22:57:36.485+0200    error   rpc error: code = Unknown desc = Docker is not available
2024-07-13T22:57:36.485+0200    error   could not list installations: could not read storage schema document: rpc error: code = Unknown desc = Docker is not available
2024-07-13T22:57:36.485+0200    error   could not list installations: could not read storage schema document: rpc error: code = Unknown desc = Docker is not available
2024-07-13T22:57:36.486+0200    debug   received EOF, stopping recv loop
2024-07-13T22:57:36.487+0200    debug   plugin process exited
2024-07-13T22:57:36.487+0200    debug   plugin exited
```

## Version
porter v1.1.0 (b50c1899)

### LINKED ISSUE #3189: Plugins logs are not controlled by the `--verbosity` flag
## Describe the bug
Plugins log level are not controlled by the `--verbosity` flag, instead it is controlled by the log files log level, e.g., `PORTER_LOGS_LEVEL`. In all other places the console log level, is controlled by the verbosity, and the code explicitly states that log file level should only affect the log level of the log file.

## To Reproduce
1. Ensure that `mongo:4.0-xenial` is **NOT** pulled
3. Run this porter command `porter list --verbosity debug`
4. See that the output doesn't contain the debug log, "pulling mongo:4.0-xenial", written by the plugin
5. Delete the `mongo:4.0-xenial` image in Docker
6. Run this porter command `PORTER_LOGS_LEVEL=debug porter list --verbosity debug`
7. See that the output now contains the debug log, "pulling mongo:4.0-xenial", written by the plugin

## Expected behavior
Plugins log level should be controlled by `--verbosity` instead of log file log level.

## Porter Command and Output
```
2024-07-13T23:08:19.432+0200    debug   Checking database schema
2024-07-13T23:08:19.432+0200    debug   Selected configured plugin      {"plugin-key": "mongodb-docker"}
2024-07-13T23:08:19.433+0200    debug   Connecting to plugin    {"plugin-command": "/home/kim/.porter/porter plugin run storage.porter.mongodb-docker"}
2024-07-13T23:08:19.433+0200    debug   starting plugin
2024-07-13T23:08:19.435+0200    debug   plugin started
2024-07-13T23:08:19.435+0200    debug   waiting for RPC address
2024-07-13T23:08:19.452+0200    debug   using plugin
2024-07-13T23:08:19.452+0200    debug   plugin address
2024-07-13T23:08:44.966+0200    debug   Initializing installation collection indices
2024-07-13T23:08:44.969+0200    debug   Initializing parameter collection indices
2024-07-13T23:08:44.969+0200    debug   Initializing credentials collection indices
-----------------------------------------------------
  NAMESPACE  NAME  VERSION  STATE  STATUS  MODIFIED
-----------------------------------------------------
2024-07-13T23:08:44.974+0200    debug   received EOF, stopping recv loop
2024-07-13T23:08:44.975+0200    debug   plugin process exited
2024-07-13T23:08:44.975+0200    debug   plugin exited
```
```
PORTER_LOGS_STRUCTURED=true PORTER_LOGS_LEVEL=debug porter list --verbosity debug
2024-07-13T23:02:59.799+0200    debug   Checking database schema
2024-07-13T23:02:59.799+0200    debug   Selected configured plugin      {"plugin-key": "mongodb-docker"}
2024-07-13T23:02:59.799+0200    debug   Connecting to plugin    {"plugin-command": "/home/kim/.porter/porter plugin run storage.porter.mongodb-docker"}
2024-07-13T23:02:59.799+0200    debug   starting plugin
2024-07-13T23:02:59.799+0200    debug   plugin started
2024-07-13T23:02:59.800+0200    debug   waiting for RPC address
2024-07-13T23:02:59.818+0200    debug   using plugin
2024-07-13T23:02:59.818+0200    debug   plugin address
2024-07-13T23:02:59.879+0200    debug   pulling mongo:4.0-xenial
2024-07-13T23:03:19.308+0200    debug   running a mongo server in a container on port 27018
2024-07-13T23:03:19.308+0200    debug   waiting for the mongo service to be ready
2024-07-13T23:03:25.172+0200    debug   Initializing installation collection indices
2024-07-13T23:03:25.174+0200    debug   Initializing parameter collection indices
2024-07-13T23:03:25.175+0200    debug   Initializing credentials collection indices
-----------------------------------------------------
  NAMESPACE  NAME  VERSION  STATE  STATUS  MODIFIED
-----------------------------------------------------
2024-07-13T23:03:25.182+0200    debug   received EOF, stopping recv loop
2024-07-13T23:03:25.184+0200    debug   plugin process exited
2024-07-13T23:03:25.184+0200    debug   plugin exited
```

## Version
porter v1.1.0 (b50c1899)

### PR DESCRIPTION
**Pull request by kichristensen**
# What does this change
* Makes the plugin log level controlable by the verbosity level instead of the log file level (#3189)
* Show logs from plugins with the correct level as defined by the plugin (#3188)

# What issue does it fix
Closes #3189 and closes #3188 

# Checklist
- [X] Did you write tests?
- [X] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

[contributors]: https://porter.sh/src/CONTRIBUTORS.md

### CODE REVIEW

File: pkg/portercontext/context_test.go [hunk 1]
```diff
@@ -63,3 +64,70 @@ func TestContext_LogToFile(t *testing.T) {
 		c.CompareGoldenFile("testdata/expected-output.txt", c.GetAllLogs())
 	}
 }
+
+func TestContext_PluginVerbosityLevel(t *testing.T) {
+	testcases := []struct {
+		name                 string
+		verbosityLevel       zapcore.Level
+		wantNumberOfLogLines int
+	}{
```
Comment (schristoff) [inline pkg/portercontext/context_test.go:73]: Why zapcore and not hclog?

**Comment by kichristensen** [comment]
@schristoff I think I got a mail about a comment on this PR, but cannot seem to find it again

**Review by schristoff — APPROVED** [review]

**Comment by KurtSchenk** [comment]
Hi @kichristensen,

I don't think I am seeing the fix.

-----------------------------------------------------
kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ docker run -it --network host --env PORTER_LOGS_STRUCTURED=true -v /var/run/docker.sock:/var/run/docker.sock porter:canary porter list --verbosity debug
2024-09-30T15:45:05.784Z        debug   Checking database schema
2024-09-30T15:45:05.784Z        debug   Selected default plugin {"plugin-key": "mongodb-docker"}
2024-09-30T15:45:05.784Z        debug   Connecting to plugin    {"plugin-command": "/root/.porter/porter plugin run storage.porter.mongodb-docker"}
2024-09-30T15:45:05.958Z        debug   Initializing installation collection indices
2024-09-30T15:45:05.960Z        debug   Initializing parameter collection indices
2024-09-30T15:45:05.961Z        debug   Initializing credentials collection indices
-----------------------------------------------------
  NAMESPACE  NAME  VERSION  STATE  STATUS  MODIFIED  
-----------------------------------------------------

kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ cleanup_docker_containers
[…]

kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ docker run -it --network host --env PORTER_LOGS_STRUCTURED=true -v /var/run/docker.sock:/var/run/docker.sock porter:canary porter list --verbosity debug
2024-09-30T15:45:19.870Z        debug   Checking database schema
2024-09-30T15:45:19.870Z        debug   Selected default plugin {"plugin-key": "mongodb-docker"}
2024-09-30T15:45:19.870Z        debug   Connecting to plugin    {"plugin-command": "/root/.porter/porter plugin run storage.porter.mongodb-docker"}
2024-09-30T15:45:32.236Z        debug   Initializing installation collection indices
2024-09-30T15:45:32.238Z        debug   Initializing parameter collection indices
2024-09-30T15:45:32.240Z        debug   Initializing credentials collection indices
-----------------------------------------------------
  NAMESPACE  NAME  VERSION  STATE  STATUS  MODIFIED  
-----------------------------------------------------

kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ cleanup_docker_containers

[…]

kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ docker run -it --network host --env PORTER_LOGS_STRUCTURED=true --env PORTER_LOGS_LEVEL=debug -v /var/run/docker.sock:/var/run/docker.sock porter:canary porter list
-----------------------------------------------------
  NAMESPACE  NAME  VERSION  STATE  STATUS  MODIFIED  


None of these show the debug detail like

kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ cleanup_docker_containers
[…]

kurt@DESKTOP-ML7A32M:/mnt/c/Users/kurtsc/certs/mongo_docs$ docker run -it --network host --env PORTER_LOGS_LEVEL=debug --env PORTER_LOGS_STRUCTURED=true -v /var/run/docker.sock:/var/run/docker.sock porter:v1.1.0 porter list --verbosity debug
2024-09-30T15:47:18.788Z        debug   Checking database schema
2024-09-30T15:47:18.788Z        debug   Selected default plugin {"plugin-key": "mongodb-docker"}
2024-09-30T15:47:18.788Z        debug   Connecting to plugin    {"plugin-command": "/root/.porter/porter plugin run storage.porter.mongodb-docker"}
2024-09-30T15:47:18.789Z        debug   starting plugin
2024-09-30T15:47:18.791Z        debug   plugin started
2024-09-30T15:47:18.791Z        debug   waiting for RPC address
2024-09-30T15:47:18.827Z        debug   using plugin
2024-09-30T15:47:18.828Z        debug   plugin address
2024-09-30T15:47:18.949Z        debug   pulling mongo:4.0-xenial
2024-09-30T15:47:28.563Z        debug   running a mongo server in a container on port 27018
2024-09-30T15:47:28.563Z        debug   waiting for the mongo service to be ready
2024-09-30T15:47:30.620Z        debug   Initializing installation collection indices
2024-09-30T15:47:30.622Z        debug   Initializing parameter collection indices
2024-09-30T15:47:30.623Z        debug   Initializing credentials collection indices
-----------------------------------------------------
  NAMESPACE  NAME  VERSION  STATE  STATUS  MODIFIED  
-----------------------------------------------------
2024-09-30T15:47:30.627Z        debug   received EOF, stopping recv loop
2024-09-30T15:47:30.629Z        debug   plugin process exited
2024-09-30T15:47:30.629Z        debug   plugin exited

**Comment by kichristensen** [comment]
@KurtSchenk Based on the output I have to agree. I will take a look

**Comment by kichristensen** [comment]
@KurtSchenk I have found the error, and is working on a fix, fully covered by tests. Thank you very much for trying it out.

**Pull request merged**